### PR TITLE
App login page

### PR DESCRIPTION
**Pull request by alexcornier**
Access to the list of applications, classes and groups from the login page without the user being logged in

### CODE REVIEW

File: vip-core/src/main/java/fr/insalyon/creatis/vip/core/server/business/VipSessionBusiness.java [hunk 1]
```diff
@@ -187,4 +187,13 @@ private String enumerationToString(Enumeration<String> enums) {
...
     }
+
+    public Boolean isUserConnected(HttpServletRequest request) {
+        HttpSession session = request.getSession();
+        User user = (User) session.getAttribute(CoreConstants.SESSION_USER);
+        if (user != null) {
+            return true;
+        }
+        return false;
+    }
```
Comment (axlbonnet) [inline vip-core/src/main/java/fr/insalyon/creatis/vip/core/server/business/VipSessionBusiness.java:198]: Maybe we should add a condition on the cookies

File: vip-core/src/main/java/fr/insalyon/creatis/vip/core/client/view/auth/SignInTab.java [hunk 2]
```diff
@@ -282,15 +282,36 @@ private void testLayoutInfo(){
...
+        infoCodeSource = WidgetUtil.getLabel("<font size=\"3\"><b>The Virtual Imaging Platform source code: " +
+                " <a href=\"https://github.com/virtual-imaging-platform\">VIP Github</a>",20);
+        infoContactus = WidgetUtil.getLabel("<font size=\"3\"><b>This portal is exclusively dedicated to non-commercial academic use, " +
+                "as indicated in the <a href=\"https://vip.creatis.insa-lyon.fr/documentation/terms.html\">terms of use.</a> " +
+                "For commercial use, please contact us at <a href=\"mailto:vip-support@creatis.insa-lyon.fr\">" +
+                "vip-support@creatis.insa-lyon.fr</a>.</b></font>",20);
+        infoMail = WidgetUtil.getLabel("<font size=\"3\"><b>Contact: <a href=\"mailto:vip-support@creatis.insa-lyon.fr\">" +
+                "vip-support@creatis.insa-lyon.fr</a></b></font>",20);
+        infoVipNews = WidgetUtil.getLabel("<font size=\"3\"><b>VIP News: <a href=\"https://www.creatis.insa-lyon.fr/vip/news.html\">" +
+                "https://www.creatis.insa-lyon.fr/vip/news.html</a>",20);
```
Comment (axlbonnet) [inline vip-core/src/main/java/fr/insalyon/creatis/vip/core/client/view/auth/SignInTab.java:314]: Maybe we could improve the formatting.

File: vip-core/src/main/java/fr/insalyon/creatis/vip/core/server/rpc/AbstractRemoteServiceServlet.java [hunk 3]
```diff
@@ -128,7 +128,7 @@ protected void authenticateSystemAdministrator(Logger logger) throws CoreExcepti
         User user = getSessionUser();
         if (!user.isSystemAdministrator()) {
             logger.error("The user has no system administrator rights: " + user.getEmail());
-            throw new CoreException("The user has no system administrator rights.");
+            throw new CoreException("The user has no sysvipSessionBusiness.tem administrator rights.");
```
Comment (axlbonnet) [inline vip-core/src/main/java/fr/insalyon/creatis/vip/core/server/rpc/AbstractRemoteServiceServlet.java:131]: typo

File: vip-core/src/main/java/fr/insalyon/creatis/vip/core/server/rpc/AbstractRemoteServiceServlet.java [hunk 4]
```diff
@@ -149,7 +149,17 @@ protected boolean isGroupAdministrator() throws CoreException {
...
+    }
+
     protected void trace(Logger logger, String message) throws CoreException {
-        logger.info("(" + getSessionUser().getEmail() + ") " + message);
+        if (vipSessionBusiness.isUserConnected(this.getThreadLocalRequest())) {
+            java.util.logging.Logger.getLogger("log").info("test si user est connecte TRUE");
+            logger.info("(" + getSessionUser().getEmail() + ") " + message);
+        } else {
+            java.util.logging.Logger.getLogger("log").info("test si use est non connecter FALSE");
+            logger.info("(Anonymous) " + message);
```
Comment (axlbonnet) [inline vip-core/src/main/java/fr/insalyon/creatis/vip/core/server/rpc/AbstractRemoteServiceServlet.java:162]: remove logs

**Review by axlbonnet — CHANGES_REQUESTED** [review]

File: vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/ApplicationModule.java [hunk 5]
```diff
@@ -63,7 +66,14 @@ public class ApplicationModule extends Module {
     public static HashMap<String, Integer> reservedClasses;
 
      public ApplicationModule() {
-     
+         CoreModule.getHomePageActions().put(CoreConstants.HOME_ACTION_SHOW_APPLICATIONS, new Runnable() {
+             @Override
+             public void run() {
+                 Logger.getLogger("log").info("test publication module");
```
[the remark this answers is not in the document]
Comment (alexcornier) [reply vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/ApplicationModule.java:72]: Done ✔️

File: vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/view/system/application/ManageApplicationsTab.java [hunk 6]
```diff
@@ -51,29 +52,40 @@ public class ManageApplicationsTab extends AbstractManageTab {
     public ManageApplicationsTab() {
 
         super(ApplicationConstants.ICON_APPLICATION, ApplicationConstants.APP_APPLICATION, ApplicationConstants.TAB_MANAGE_APPLICATION);
+        this.setCanClose(true);
```
[the remark this answers is not in the document]
Comment (alexcornier) [reply vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/view/system/application/ManageApplicationsTab.java:55]: Nothing, i removed it ✔️

File: vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/view/system/application/ManageApplicationsTab.java [hunk 7]
```diff
@@ -51,29 +52,40 @@ public class ManageApplicationsTab extends AbstractManageTab {
...
 
-        versionInfoLayout.addMember(editVersionLayout);
-        versionInfoLayout.addMember(publishVersionLayout);
-        versionLayout.addMember(versionInfoLayout);
-        vLayout.addMember(versionLayout);
+            versionInfoLayout.addMember(editVersionLayout);
+            versionInfoLayout.addMember(publishVersionLayout);
+            versionLayout.addMember(versionInfoLayout);
+            vLayout.addMember(versionLayout);
+        }
```
[the remark this answers is not in the document]
Comment (alexcornier) [reply vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/view/system/application/ManageApplicationsTab.java:79]: Done ✔️

File: vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/bean/Application.java [hunk 8]
```diff
@@ -41,40 +43,40 @@
...
     private String owner;
     private String fullName;
+    private List<String> applicationClasses;
+    private List<String> applicationGroups;
 
     public Application() {
     }
 
-    public Application(String name, String citation) {
+    public Application(String name, List<String> classes, String owner, String citation, List<String> groups){}
```
[the remark this answers is not in the document]
Comment (alexcornier) [reply vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/bean/Application.java:55]: Done ✔️

[the remark this answers is not in the document]
Comment (axlbonnet) [reply vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/bean/Application.java:55]: It is still there.. maybe a merge glitch

File: vip-core/src/main/java/fr/insalyon/creatis/vip/core/client/view/CoreConstants.java [hunk 9]
```diff
@@ -100,6 +100,9 @@ public class CoreConstants implements IsSerializable {
     public static final String TAB_MANAGE_TIPS = "manage_tips_tab";
     public static final String TAB_MANAGE_USERS = "manage_users_tab";
     public static final String TAB_MANAGE_SETTING = "manage_setting_tab";
+    public static final String TAB_APPLICATIONS_NOT_LOG = "application_not_log";
+    // home actions
+    public static final String HOME_ACTION_SHOW_APPLICATIONS = "SHOW_PPLICATIONS";
```
[the remark this answers is not in the document]
Comment (axlbonnet) [reply vip-core/src/main/java/fr/insalyon/creatis/vip/core/client/view/CoreConstants.java:104]: It is still there.. maybe a merge glitch

**Review by axlbonnet — APPROVED** [review]

**Pull request merged**